sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/repository/support/StringBasedCosmosQuery.java

```diff
@@ -49,13 +52,29 @@ public Object execute(final Object[] parameters) {
         final CosmosParameterAccessor accessor = new CosmosParameterParameterAccessor(getQueryMethod(), parameters);
         final ResultProcessor processor = getQueryMethod().getResultProcessor().withDynamicProjection(accessor);
 
-        List<SqlParameter> sqlParameters = getQueryMethod().getParameters().stream()
-                            .filter(p -> !Pageable.class.isAssignableFrom(p.getType()) && !Sort.class.isAssignableFrom(p.getType()))
-                            .map(p -> new SqlParameter("@" + p.getName().orElse(""),
-                                                       toCosmosDbValue(parameters[p.getIndex()])))
-                            .collect(Collectors.toList());
+        String expandedQuery = query;
+        List<SqlParameter> sqlParameters = new ArrayList<>();
+        for (int paramIndex = 0; paramIndex < parameters.length; paramIndex++) {
+            Parameter queryParam = getQueryMethod().getParameters().getParameter(paramIndex);
+            if (parameters[paramIndex] instanceof Collection) {
+                ArrayList<String> expandParam = (ArrayList<String>) ((Collection<?>) parameters[paramIndex]).stream()
+                    .map(Object::toString).collect(Collectors.toList());
+                List<String> expandedParamKeys = new ArrayList<>();
+                for (int arrayIndex = 0; arrayIndex < expandParam.size(); arrayIndex++) {
+                    String paramName = "@" + queryParam.getName().orElse("") + arrayIndex;
+                    expandedParamKeys.add(paramName);
+                    sqlParameters.add(new SqlParameter(paramName, toCosmosDbValue(expandParam.get(arrayIndex))));
+                }
+                expandedQuery = expandedQuery.replaceAll("@" + queryParam.getName().orElse(""), String.join(",", expandedParamKeys));
+            } else {
+                if (!Pageable.class.isAssignableFrom(queryParam.getType())
+                    && !Sort.class.isAssignableFrom(queryParam.getType())) {
+                    sqlParameters.add(new SqlParameter("@" + queryParam.getName().orElse(""), toCosmosDbValue(parameters[paramIndex])));
+                }
```

Comment:
For better understanding for future, we should add a comment here on the else condition, where the sqlParameter will remain an empty array list.

Reply:
@kushagraThapar  I am not actually sure why that is there, I carried it over from the past functionality, but it seems odd to me, as if we set the parameter name to "@" alone it will surely break. Do you know why it is there/what the comment should be?

Reply:
@trande4884 - what I meant was in the section of the code, where there is no else condition and the code will eventually create an empty `sqlParameter` list.

```
if (!Pageable.class.isAssignableFrom(queryParam.getType())
                    && !Sort.class.isAssignableFrom(queryParam.getType())) {
                    sqlParameters.add(new SqlParameter("@" + queryParam.getName().orElse(""), toCosmosDbValue(parameters[paramIndex])));
                }
```

The reason it is here is because page query and sort queries are handled down below. so that's why we filter them here. But on second thought, I don't think the comment is really necessary, I think we can skip it and go with the implementation as it is.